packages/mui-joy/src/colorInversion/colorInversionUtils.spec.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { styled } from '@mui/joy/styles';
+import { applySoftInversion, applySolidInversion } from '@mui/joy/colorInversion';
+import Box from '@mui/joy/Box';
+
+/**
+ * sx prop type check
+ */
+<Box sx={[(theme) => ({}), applySoftInversion('neutral'), applySoftInversion('primary')]} />;
+
+<Box
+  sx={[
+    (theme) => ({}),
+    // @ts-expect-error no `unknown` color from theme palette
+    applySoftInversion('unknown'),
```

Comment:
Does it work with custom colors too if developers add them in the theme?

Reply:
Yes, they can add more color by augmenting the `ColorPalettePropOverrides` from the theme.